app/models/widget.rb

```diff
@@ -0,0 +1,21 @@
+class Widget < ActiveRecord::Base
+
+  belongs_to :user
+
+  validates :template, :presence => true
+  validates :name,     :presence => true
+
+  serialize :data
+
+  before_save :default_values
+
+  def default_values
+    self.sizex ||= 4
```

Comment:
maybe add to validation, so users cant put any value? e.g. validate numerically, greater then 0, smaller then.. ?